Make code workflow-compatible, including automated container build (#6)
hdr-bgnn/drexel_metadata: README.md

7. Fish body must not be folded and should have no curvature.

These do not need to be adhered to if properly set up/modified for a specific use case.
The model is available on data commons osc as "Drexel_metadata_generator" at https://datacommons.tdai.osu.edu/dataverse/fish-traits/ 

> However this model is not published yet. To download you need an acount on datacommons.tdai.osu and you will need to create a DATAVERS_API_TOKEN 

Reviewer: ```suggestion
However this model is not published yet. To download you will need an account on datacommons.tdai.osu and you will need to create an API Token and set the  DATAVERSE_API_TOKEN environment variable.
```

Reviewer 2: We should clarify the following things first:
1. Can the model be made public at this point? If yes, we should make it so, and also consider putting it on [Huggingface](https://huggingface.co/imageomics). If not, what are the criteria for making it public.
2. If the model can't be public yet, the description above won't cut it, because people will need not only an account put also be given access.

Author: 1- make the model public, i say yes but that is not my call, the ball is in Jane Greenberg (@hlapp, you were cc in the email, the response from Jane was unclear). for huggingface, yes that is next on my todo list. 

2- Sure but keep it mind that right now, the point (in priority) is to make this project available and documented for people from our institution BGNN/imageomics. So I assume that people will have (or should have) access to the datacommons/dataverse.

and you can use :
```

> python dataverse_download.py https://covid-commons.osu.edu/ doi:10.5072/FK2/MMX6FY output/

Reviewer: ```suggestion
python dataverse_download.py https://datacommons.tdai.osu.edu/ doi:10.5072/FK2/MMX6FY output/
```

the number of files processed in the directory. The `limit` positional argument is only applicable when passing a directory. 

#### Device Configuration
By default `gen_metadata.py` requires a GPU (cuda).
To use a CPU instead pass the `--device cpu` argument to `gen_metadata.py`.

#### Single File Usage
The following three arguments are only supported when processing a single image file:
- `--outfname <filename>` - When passed the script will save the output metadata JSON to `<filename>` instead of printing to the console (the default behavior when processing one file).
- `--maskfname <filename>` - Enables logic to save an output mask to `<filename>` for the single input file.
- `--visfname <filename>` - Changes the script to save the output visualization to `<filename>` instead of the hard coded location.

These arguments are meant to simplify adding `gen_metadata.py` to a workflow that process files individually.

### Running with Singularity
A Docker container is automatically built for each **drexel_metadata** release. This container has the requirements installed and includes the model file.
To run the singularity container for a specific version follow this pattern:
```

> singularity run docker://ghcr.io/hdr-bgnn/drexel_metadata:<release> python gen_metadata.py ...

Reviewer 2: Are you sure this shouldn't be `python3` instead of `python`? And the usage command above doesn't prefix with the interpreter, so maybe we don't need to here, either, which would be naturally consistent with the usage command above.

| solidity             | Per Fish                 | Float             | The ratio of pixels in the fish to pixels of the convex hull image.                                                              |
| std             | Per Fish                 | Float             | The standard deviation of the mask pixel coordinate distribution. |

## Associated Publication

> [Joel Pepper et al.](https://ieeexplore.ieee.org/document/9651834)

Reviewer 2: ```suggestion
J. Pepper, J. Greenberg, Y. Bakiş, X. Wang, H. Bart and D. Breen, "Automatic Metadata Generation for Fish Specimen Image Collections," 2021 ACM/IEEE Joint Conference on Digital Libraries (JCDL), 2021, pp. 31-40, doi: [10.1109/JCDL52503.2021.00015](https://doi.org/10.1109/JCDL52503.2021.00015).
```

> [Kevin Karnani et al.] (https://assets.researchsquare.com/files/rs-1506561/v1_covered.pdf?c=1651071974)

Reviewer 2: ```suggestion
Kevin Karnani, Joel Pepper, Yasin Bakis et al. Computational Metadata Generation Methods for Biological Specimen Image Collections, 27 April 2022, PREPRINT (Version 1) available at Research Square <https://doi.org/10.21203/rs.3.rs-1506561/v1>
```
